fabric-item-api-v1/src/main/java/net/fabricmc/fabric/impl/item/DataComponentTooltipsImpl.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.item;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import net.minecraft.component.ComponentType;
+import net.minecraft.item.tooltip.TooltipAppender;
+
+public final class DataComponentTooltipsImpl {
+	private static final List<ComponentType<? extends TooltipAppender>> types = new ArrayList<>();
+
+	private DataComponentTooltipsImpl() {
+	}
+
+	public static void register(ComponentType<? extends TooltipAppender> type) {
+		if (!types.contains(type)) {
+			types.add(type);
+		} else {
```

Comment:
Use the early return pattern, it will make the code much easier to read. E.g

```
if (type.contains(type)) {
  throw new ...
}

types.add(...
```